querybook/server/logic/comment.py

```diff
@@ -0,0 +1,147 @@
+from querybook.server.app.db import with_session
+from models.comment import Comment, CommentReaction, DataTableComment, DataCellComment
+from querybook.server.const.comment import CommentDict
+
+
+@with_session
+def get_comment_by_id(comment_id: int, session=None):
+    return Comment.get(id=comment_id, session=session)
+
+
+@with_session
+def get_all_comment_data_dict_by_id(comment_id: int, session=None) -> CommentDict:
+    comment = Comment.get(id=comment_id, session=session)
+    child_comments = (
+        session.query(Comment)
+        .filter(Comment.parent_comment_id == comment_id)
+        .order_by(Comment.created_at.asc())
+        .all()
+    )
+    reactions = (
+        session.query(CommentReaction)
+        .filter(CommentReaction.comment_id == comment_id)
+        .all()
+    )
+    comment_dict = comment.to_dict()
+    if child_comments:
+        comment_dict["child_comments"] = [
+            child_comment.to_dict() for child_comment in child_comments
+        ]
+    comment_dict["reactions"] = [reaction.to_dict() for reaction in reactions]
+
+    return comment_dict.sort(key=lambda element: element["created_at"])
+
+
+@with_session
+def get_comments_by_data_cell_id(data_cell_id: int, session=None):
+    cell_comments = (
+        session.query(DataCellComment)
+        .filter(DataCellComment.data_cell_id == data_cell_id)
+        .all()
+    )
+
+    comments = []
+    for cell_comment in cell_comments:
+        comment_dict = get_all_comment_data_dict_by_id(
+            cell_comment.comment_id, session=session
+        )
+        comments.append(comment_dict)
+
+    return comments
```

Comment:
use list comprehension to simplify 
comments = [get_all_comment_data_dict_by_id(cell_comment.comment_id, session=session) for cell_comment in cell_comments]